community/kernel/src/main/java/org/neo4j/kernel/impl/proc/Procedures.java

```diff
@@ -217,4 +219,25 @@ public void start() throws Throwable
         // And register built-in procedures
         builtin.accept( this );
     }
+
+    private boolean changed = false;
+
+    public void writerCreateToken( boolean allow )
+    {
+        if ( !changed )
```

Comment:
This seems to be protecting against people abusing the dependency resolver to enable token writes from within procedures. However, that assumes access to GraphDatabaseAPI which already allows them to bypass security entirely, so we are perhaps protecting unnecessarily. Better to remove access to GraphDatabaseAPI. Although that can be part of another PR.